core/drivers/clk/clk-stm32mp21.c

```diff
@@ -0,0 +1,3781 @@
+							  RCC_CIDCFGR(i));
+
+	for (i = 0; i < RCC_NB_CONFS; i++) {
+		pdata->conf_data.priv_conf[i] = io_read32(pdata->rcc_base +
+							  RCC_PRIVCFGR(i));
+		pdata->conf_data.sec_conf[i] = io_read32(pdata->rcc_base +
+							 RCC_SECCFGR(i));
+		pdata->conf_data.lock_conf[i] = io_read32(pdata->rcc_base +
+							  RCC_RCFGLOCKR(i));
+		pdata->conf_data.access_mask[i] = GENMASK_32(31, 0);
+	}
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result stm32_rcc_rif_pm(enum pm_op op, unsigned int pm_hint,
+				   const struct pm_callback_handle *h __unused)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	bool is_tdcid = false;
+
+	if (!PM_HINT_IS_STATE(pm_hint, CONTEXT))
+		return TEE_SUCCESS;
+
+	res = stm32_rifsc_check_tdcid(&is_tdcid);
+	if (res)
+		return res;
+
+	if (op == PM_OP_RESUME) {
+		if (is_tdcid)
+			res = stm32_rcc_rif_pm_resume();
+		else
+			res = handle_available_semaphores();
+	} else {
+		if (!is_tdcid)
+			return TEE_SUCCESS;
+
+		res = stm32_rcc_rif_pm_suspend();
+	}
+
+	return res;
+}
+
+static TEE_Result rcc_rif_config(void)
+{
+	TEE_Result res = TEE_ERROR_ACCESS_DENIED;
+	bool is_tdcid = false;
+
+	/* Not expected to fail at this stage */
+	res = stm32_rifsc_check_tdcid(&is_tdcid);
+	if (res)
+		panic();
+
+	res = apply_rcc_rif_config(is_tdcid);
+	if (res)
+		panic();
+
+	register_pm_core_service_cb(stm32_rcc_rif_pm, NULL, "stm32-rcc-rif");
+
+	return TEE_SUCCESS;
+}
+
+driver_init_late(rcc_rif_config);
+
+static TEE_Result stm32mp2_clk_probe(const void *fdt, int node,
+				     const void *compat_data __unused)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	int fdt_rc = 0;
+	int rc = 0;
+	struct clk_stm32_priv *priv = &stm32mp21_clock_data;
+	struct stm32_clk_platdata *pdata = &stm32mp21_clock_pdata;
+	int subnode = 0;
+
+	fdt_rc = stm32_clk_parse_fdt(fdt, node, pdata);
+	if (fdt_rc) {
+		EMSG("Failed to parse clock node: %d", fdt_rc);
+		return TEE_ERROR_GENERIC;
+	}
+
+	fdt_for_each_subnode(subnode, fdt, node) {
+		res = dt_driver_maybe_add_probe_node(fdt, subnode);
+		if (res) {
+			EMSG("Failed on node %s with %#"PRIx32,
+			     fdt_get_name(fdt, subnode, NULL), res);
+			goto err;
+		}
+	}
+
+	if (IS_ENABLED(CFG_STM32_CM33TDCID)) {
+		res = handle_available_semaphores();
+		if (res)
+			return res;
+	}
+
+	rc = clk_stm32_init(priv, stm32_rcc_base());
+	if (rc) {
+		res = TEE_ERROR_GENERIC;
+		goto err;
+	}
+
+	stm32mp2_init_clock_tree(priv, pdata);
+
+	clk_stm32_init_oscillators(fdt, node);
+
+	res = clk_stm32_apply_rcc_config(pdata);
+	if (res)
+		panic("Error when applying RCC config");
+
+	stm32mp_clk_provider_probe_final(fdt, node, priv);
+
+	if (IS_ENABLED(CFG_STM32_CLK_DEBUG))
+		clk_print_tree();
+
+	return TEE_SUCCESS;
+err:
+	free(pdata->conf_data.cid_confs);
+	free(pdata->conf_data.sec_conf);
+	free(pdata->conf_data.priv_conf);
+	free(pdata->conf_data.lock_conf);
+	free(pdata->conf_data.access_mask);
+
+	return res;
+}
+
+CLK_DT_DECLARE(stm32mp21_clk, "st,stm32mp21-rcc", stm32mp2_clk_probe);
```

Comment:
Nitpicking: rename `stm32mp2_clk_probe()` to `stm32mp21_clk_probe()` for consistency?